esphome/components/tc74/tc74.cpp

```diff
@@ -0,0 +1,72 @@
+// Based on the TC74 datasheet https://ww1.microchip.com/downloads/en/DeviceDoc/21462D.pdf
+
+#include "tc74.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace tc74 {
+
+static const char *const TAG = "tc74";
+
+static const uint8_t TC74_REGISTER_TEMPERATURE = 0x00;
+static const uint8_t TC74_REGISTER_CONFIGURATION = 0x01;
+static const uint8_t TC74_DATA_READY_MASK = 0x40;
+
+// It is possible the "Data Ready" bit will not be set if the TC74 has not been powered on for at least 250ms, so it not
+// being set does not constitute a failure.
+void TC74Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up TC74...");
+  uint8_t config_reg;
+  if (this->read_register(TC74_REGISTER_CONFIGURATION, &config_reg, 1) != i2c::ERROR_OK) {
+    this->mark_failed();
+    return;
+  }
+  this->data_ready_ = config_reg & TC74_DATA_READY_MASK;
+}
+
+void TC74Component::update() {
+  this->set_timeout("temperature", 2, [this]() { this->read_temperature_(); });
```

Comment:
I don't see the point in delaying the actual read by 2ms here (which will actually be more like 16ms as it wont execute until the next `loop` comes around and runs the next scheduled tasks).
```suggestion
  this->read_temperature_();
```